testng/src/main/java/com/github/automatedowl/tools/JSErrorsCollectorListener.java

```diff
@@ -54,17 +54,17 @@ public void afterInvocation(IInvokedMethod iInvokedMethod, ITestResult iTestResu
             // Check for JS errors assertion boolean flag.
             // Skip error throwing in case of negative unit test.
             if (isAssertJSErrorsEnabled(
-                    iInvokedMethod) && getJSErrorsFromLogEntries(logEntries).anyMatch(e -> true)
+                    iInvokedMethod) && getJSErrorsFromLogEntries(logEntries).count()!=0
                     && iInvokedMethod
                     .getTestMethod().getTestClass().toString().contains("com.github.automatedowl")
                     && iInvokedMethod
                     .getTestMethod().getConstructorOrMethod().getMethod().getName().equals("referenceErrorTest")) {
                 // Don't throw exception on unit test.
             }
             else if (isAssertJSErrorsEnabled(
-                    iInvokedMethod) && getJSErrorsFromLogEntries(logEntries).anyMatch(e -> true)) {
+                    iInvokedMethod) && getJSErrorsFromLogEntries(logEntries).count()!=0) {
                 logger.severe(JS_ERRORS_EXCEPTION_STRING);
-                iTestResult.setStatus(ITestResult.FAILURE);
+                Assert.fail(getJSErrorsFromLogEntries(logEntries).map(LogEntry::getMessage).collect(Collectors.joining("\n")));
```

Comment:
Have you made sure that this addition works well with closing the driver in afterMethod call? @mkulikov

Reply:
yes, I'm sure